### DISCUSSION TITLE
New install, LSP display issue

### DISCUSSION THREAD

**Discussion by resistorsoftware** [LazyVim/LazyVim#5084] [General]
I used Mason to install ruby-lsp. I then navigated to a Ruby file. I then moved my cursor to a class name, and used the <leader>cS to initiate the LSP References and Definitions help. It worked but the window that opens to display results is a window in a vertical split, aligned far right, and taking up about 10% of the screen in the terminal and neovim. 

Where would that split be getting fed information about how big it should be when it renders? I would think the split should be more like half the current open buffer? Any tips on this?

**Comment by dpetka2001** [comment]
https://github.com/folke/trouble.nvim/discussions/394 and https://github.com/folke/trouble.nvim/discussions/506 should be of help

**Reply by resistorsoftware** [reply]
Unfortunately overriding the trouble settings in options.lua by returning a table with mode changes, window sizes has done zip, zero, zilch on this window split opening with *Definitions* and *References*. It is as if the trouble plugin is no longer responsible for the display. I am sure it is something silly I am doing, but the search for that continues! Thanks for the tip though, on trying to set a window size.

**Reply by dpetka2001** [reply]
Can you try running the command `:Trouble lsp_document_symbols win.position=right win.size=40` and see if that works for you? Change the number to whatever you want.

**Reply by resistorsoftware** [reply]
Yes! That does the trick! If I changed to win.size=80 it was even nicer... so now I just need to learn how to properly configure this. Clearly it is my skill issue in trying to teach the trouble config what I would like.

**Reply by dpetka2001** [reply]
You can probably just change the default keymap to do what you want. So, something like
```lua
return {
  {
    "folke/trouble.nvim",
    keys = {
      { "<leader>cs", "<cmd>Trouble lsp_document_symbols win.position=right win.size=80<cr>", desc = "Symbols (Trouble)" },
    },
  },
}
```

**Comment by resistorsoftware** [comment]
So you pointed to a mode there of *lsp_document_symbols* and I see from the example *diagnostics* is used a lot. My little window that keeps opening small is for *definitions* and *references* so I am wondering what mode that might be? 

In other words, while the lsp_document_symbols for Trouble window opens fine, my desire is on the <leader>cS options which come from examining the symbol under the cursor and then finding the definitions and references. For this window, I am not sure what that is in terms of Trouble, and how to inform that window of info, hey! Draw yourself out nicer!

**Reply by dpetka2001** [reply]
It's the same mode, but it shows less symbols than the default call to `lsp_document_symbols` would show. You can see what symbols it shows [here](https://github.com/folke/trouble.nvim/blob/46cf952fc115f4c2b98d4e208ed1e2dce08c9bf6/README.md?plain=1#L238-L267). Feel free to use whichever you prefer better.

**Reply by dpetka2001** [reply]
Try the command `:Trouble symbols win.position=right win.size=100` and see for yourself. And then use that in your keybinding if you prefer that.

**Comment by resistorsoftware** [comment]
I found that what I wanted was to run `:Trouble lsp` to reveal the definitions and references! Now to figure out how to get that in a nice pop-up floating window!

**Reply by dpetka2001** [reply]
Ohhh, you meant `<leader>cS` (capital S). I didn't read carefully. Sorry about that.

Yes, in that case `lsp` is what you want and you can try the command `:Trouble lsp win.type=float`

**Reply by dpetka2001** [reply]
And if you want also border (to make it more distinguishable) and to automatically jump to the float window then try `Trouble lsp win.type=float win.border=rounded focus=true`

**Reply by resistorsoftware** [reply]
That is it! That is how I wanted the <leader>cS to be. I am coming from a different keymapping of gD (goto definition) but I can use cS as well. Anyway to decorate the floating windows? Super nice stuff! Thank you so much for helping me out !

**Reply by dpetka2001** [reply]
Please see my previous comment about adding border to the float window.

**Reply by dpetka2001** [reply]
So, 
```lua
return {
  {
    "folke/trouble.nvim",
    keys = {
      { "<leader>gD", "<cmd>Trouble lsp win.type=float win.border=rounded focus=true<cr>", desc = "Symbols (Trouble)" },
    },
  },
}
```
if you want `<leader>gD` to be your keymap.

PS: edited the correct command and not the first one that I posted.

**Reply by resistorsoftware** [reply]
I put that into `lua/config/options.lua` but nothing changed in the use :)

I thought that was how to override the defaults, but apparently not.

**Reply by dpetka2001** [reply]
No, `options.lua` is to set Neovim options (like `shiftwidth`, `foldexpr` and other global variables). Put the above in `~/.config/nvim/lua/plugins/trouble.lua` (the name of the file can be whatever you want it doesn't really matter). Every customization you want to do for a plugin, you do it in a file under `~/.config/nvim/lua/plugins/` directory.

**Reply by resistorsoftware** [reply]
Learning good things. Thanks!
